From 57dda46efe65d800a7ab21a05603f35031c5a341 Mon Sep 17 00:00:00 2001
From: Shaun Tancheff <shaun@tancheff.com>
Date: Mon, 18 Apr 2016 08:11:49 -0500
Subject: [PATCH 93/93] dm-io: use on-stack plugging

Add a plug to dispatch_io() to avoid having the submitted
I/O broken up during submission.

Signed-off-by: Hannes Reinecke <hare@suse.com>

Modified for v4.6.0-rc4 / bio split flags
Signed-off-by: Shaun Tancheff <shaun.tancheff@seagate.com>
---
 drivers/md/dm-io.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/md/dm-io.c b/drivers/md/dm-io.c
index f96ba41..4b958aa 100644
--- a/drivers/md/dm-io.c
+++ b/drivers/md/dm-io.c
@@ -292,6 +292,7 @@ static void do_region(int op, int op_flags, unsigned region,
 	unsigned short logical_block_size = queue_logical_block_size(q);
 	sector_t num_sectors;
 	unsigned int uninitialized_var(special_cmd_max_sectors);
+	struct blk_plug plug;
 
 	/*
 	 * Reject unsupported discard and write same requests.
@@ -306,6 +307,7 @@ static void do_region(int op, int op_flags, unsigned region,
 		return;
 	}
 
+	blk_start_plug(&plug);
 	/*
 	 * where->count may be zero if op holds a flush and we need to
 	 * send a zero-sized flush.
@@ -361,6 +363,7 @@ static void do_region(int op, int op_flags, unsigned region,
 		atomic_inc(&io->count);
 		submit_bio(bio);
 	} while (remaining);
+	blk_finish_plug(&plug);
 }
 
 static void dispatch_io(int op, int op_flags, unsigned int num_regions,
-- 
2.8.0.rc3

